canonical-web-and-design/jaas-dashboard
# Fix rendering of local charm name and icon

## Done

When a model contained a local charm it would break the rendering because it was receiving the name in an incorrect format. This renders a default charm icon when it's a local charm for the time being and fixes a styling issue with the closed sidebar.

## QA

- Bootstrap a controller on a public cloud using the juju rc2 snap.
- Deploy a local charm.
 - The one used in testing `https://github.com/dshcherb/cockroachdb-operator`
 - Clone that repo.
 - Then from within the cloned directory `juju deploy .`
- `juju dashboard` and view the built in dashboard to accept the self-signed cert.
- Pull code
- Open config.js and change: 
  - `baseControllerURL` to the ip address and port (if any) from above.
  - `isJuju` to `true`
  - `identityProviderAvailable` to `false`
- `./run`
- Open http://0.0.0.0:8036/
- View the default model, it should show you the model details as per the design below.

## Details

Fixes #515 
Fixes #527
Fixes #534

## Screenshots

![Screen Shot 2020-05-29 at 3 56 41 PM](https://user-images.githubusercontent.com/532033/83309467-94c8b700-a1c6-11ea-9b36-84bc0a9c504d.png)
